### DISCUSSION TITLE
Ambiguous Shortcut: loc

### DISCUSSION THREAD

**Discussion by IGLADI** [searxng/searxng#3569] [Q&A] [answered]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
Dockered version (pulled from searxng/searxng:latest): 2024.2.25+c0b97c654 (the bug comes both with the base version & a fork I use)

**How did you install SearXNG?**
Docker Compose, from searxng/searxng:latest

**What happened?**
I'm trying to play with the "private special engines" (like find, locate) and don't really find any documentation on them so already if there is somewhere with everything explained + how to protect them on public instances I would be thankfull)
But when I uncomment the "locate" engine (taken from `settings.yml` out of the repo and only enabled it:
```    # Be careful when enabling this engine if you are
    # running a public instance. Do not expose any sensitive
    # information. You can restrict access by configuring a list
    # of access tokens under tokens.
    - name: locate
      engine: command
      command: ["locate", "{{QUERY}}"]
      shortcut: loc
      tokens: []
      disabled: false
      delimiter:
          chars: " "
          keys: ["line"]
```
Searxng simply stops working (see screenshot) and I get a 504 (gateway timout as it is behind a reverse proxy and it never answers)
**After debugging I found out that if you remove the shortcut loc it indeed just works. (I took this form the repos `settings.yml`+ the shortcut isn't used anywhere else in my settings)**
It seems that !loc by default redirects me to the picture of `https://www.loc.gov/` ?

**How To Reproduce**
Use the settings.yml from the repo and uncomment the locate engine (+enable it) and try accessing searxng.

**Expected behavior**
Just be able to go on searxng (I haven't had time to check how those features work I just found out this bug when setting those up)

**Screenshots & Logs**
![image](https://github.com/searxng/searxng/assets/128155180/4a7284a2-4d2c-41f5-96be-02474e7a54cd)
I also get this log:
`2024-06-12 14:30:35,075 ERROR:searx.engines: Engine config error: ambiguous shortcut: loc`
When it's the only one:
![image](https://github.com/searxng/searxng/assets/128155180/90dc276e-e70e-484b-921b-72e832ec6dfd)
And then it just crashes:
![image](https://github.com/searxng/searxng/assets/128155180/468c4caf-ccc9-4ce2-b18f-fdddff119275)

**Comment by IGLADI** [comment]
Ow and those engines seem completely broken in docker because find and other commands aren't downloaded in the dockerfile?:
![image](https://github.com/searxng/searxng/assets/128155180/1992d5a0-0471-4614-a694-e80640fcd834)

Can't find any docs on how to download them inside the docker container with some setting neither how to use them or anything. (like in docker find won't have access to my folders, so should I map my whole filesystem in a particular folder or smth?)

**Reply by return42** [reply, marked as answer]
> and don't really find any documentation on them so already if there is somewhere with everything explained + how to protect them on public instances I would be thankfull

Read [Command Line Engines](https://docs.searxng.org/dev/engines/offline/command-line-engines.html#command-line-engines) and follow link to [Private Engines (tokens)](https://docs.searxng.org/admin/settings/settings_engine.html#private-engines).

> It seems that !loc by default redirects me to the picture of https://www.loc.gov/ ?

Yes, `!loc` is already used by:

https://github.com/searxng/searxng/blob/f5eb56b63f250c7804e5e1cf4426e550bc933906/searx/settings.yml#L1072-L1075
